models/org_team.go

```diff
@@ -34,6 +35,75 @@ type Team struct {
 	Units       []*TeamUnit `xorm:"-"`
 }
 
+// SearchTeamOptions holds the search options
+type SearchTeamOptions struct {
+	UserID      int64
+	Keyword     string
+	OrgID       int64
+	IncludeDesc bool
+	PageSize    int
+	Page        int
+}
+
+// SearchTeam search for teams. Caller is responsible to check permissions.
+func SearchTeam(opts *SearchTeamOptions) ([]*Team, int64, error) {
+	if opts.Page <= 0 {
+		opts.Page = 1
+	}
+	if opts.PageSize == 0 {
+		// Default limit
+		opts.PageSize = 10
+	}
+
+	var cond = builder.NewCond()
+
+	if len(opts.Keyword) > 0 {
+		lowerKeyword := strings.ToLower(opts.Keyword)
+		var keywordCond = builder.NewCond()
+		if opts.IncludeDesc {
+			keywordCond = builder.Or(
+				builder.Like{"lower_name", lowerKeyword},
+				builder.Like{"LOWER(description)", lowerKeyword},
+			)
+		} else {
+			keywordCond = builder.Or(
+				builder.Like{"lower_name", lowerKeyword},
+			)
+		}
+		cond = cond.And(keywordCond)
+	}
+
+	cond = cond.And(builder.Eq{"org_id": opts.OrgID})
+
+	sess := x.NewSession()
+	defer sess.Close()
+
+	count, err := sess.
+		Where(cond).
+		Count(new(Team))
+
+	if err != nil {
+		return nil, 0, err
+	}
+
+	sess = sess.Where(cond)
+	if opts.PageSize > 0 {
+		sess = sess.Limit(opts.PageSize, (opts.Page-1)*opts.PageSize)
+	}
+	if opts.PageSize == -1 {
```

Comment:
`else`?

Reply:
I think it was covered already, but I restructured to make it clearer.